src/main/java/io/fabric8/jenkins/openshiftsync/ConfigMapWatcher.java

```diff
@@ -111,70 +109,32 @@ public void start() {
-                        // that a PodTemplate uses, we just
-                        // recreate the PodTemplate altogether. This makes it so
-                        // that any changes from within
-                        // Jenkins is undone.
-                        for (PodTemplate podTemplate : trackedConfigMaps
-                                .get(configMap.getMetadata().getUid())) {
-                            JenkinsUtils.removePodTemplate(podTemplate);
-                        }
-
-                        for (PodTemplate podTemplate : podTemplatesFromConfigMap(configMap)) {
-                            JenkinsUtils.addPodTemplate(podTemplate);
-                        }
-                    } else {
-                        // The user modified the configMap to no longer be a
-                        // jenkins-slave.
-                        for (PodTemplate podTemplate : trackedConfigMaps
-                                .get(configMap.getMetadata().getUid())) {
-                            JenkinsUtils.removePodTemplate(podTemplate);
-                        }
-
-                        trackedConfigMaps.remove(configMap.getMetadata()
-                                .getUid());
-                    }
-                } else {
-                    if (containsSlave(configMap)) {
-                        // The user modified the configMap to be a jenkins-slave
-
-                        List<PodTemplate> templates = podTemplatesFromConfigMap(configMap);
-                        trackedConfigMaps.put(configMap.getMetadata().getUid(),
-                                templates);
-                        for (PodTemplate podTemplate : templates) {
-                            JenkinsUtils.addPodTemplate(podTemplate);
-                        }
-                    }
-                }
+                processSlavesForModifyEvent(slavesFromCM, cmType, uid, cmname, namespace);
                 break;
 
             case DELETED:
-                if (trackedConfigMaps.containsKey(configMap.getMetadata()
-                        .getUid())) {
-                    for (PodTemplate podTemplate : trackedConfigMaps
-                            .get(configMap.getMetadata().getUid())) {
-                        JenkinsUtils.removePodTemplate(podTemplate);
+                if (trackedPodTemplates.containsKey(uid)) {
+                    for (PodTemplate podTemplate : trackedPodTemplates.get(uid)) {
+                        String type = podTemplateToApiType.get(podTemplate.getName());
+                        if (type != null && type.equals(cmType)) {
+                            podTemplateToApiType.remove(podTemplate.getName());
+                            JenkinsUtils.removePodTemplate(podTemplate);
+                        }
                     }
-                    trackedConfigMaps.remove(configMap.getMetadata().getUid());
+                    trackedPodTemplates.remove(uid);
```

Comment:
Why not call `processSlavesForDeleteEvent` ?

Reply:
cause I meant to but forgot :-) 

part of next push / thx